typing/types.ml

```diff
@@ -454,6 +454,15 @@ let decl_is_abstract decl =
   | Type_abstract _ -> true
   | Type_record _ | Type_variant _ | Type_open -> false
 
+let decl_is_unboxed decl =
+  match decl.type_kind with
+    Type_record ([{ld_type = arg; _}], Record_unboxed _)
+  | Type_variant ([{cd_args = Cstr_tuple [arg]; _}], Variant_unboxed)
+  | Type_variant ([{cd_args = Cstr_record [{ld_type = arg; _}]; _}],
+                  Variant_unboxed) ->
+    Some arg
+  | _ -> None
```

Comment:
I would prefer to avoid a fragile pattern at the end here, by listing all the constructors instead. This would force us to revisit the definition if we add a new kind later.

Reply:
Agreed.  I've updated this to list the constructors.  I stopped short of making the new pattern completely non-overlapping with the previous one (which would require also a bunch of matches on the lengths of lists), but let me know if you want that.

Reply:
(I'm okay with overlapping patterns, I just like to make sure that we are forced to consider this code again if we add a new constructor to the datatype.)